contracts/CollectionParty.sol

```diff
@@ -66,21 +73,26 @@ contract PartyBuy is Party {
 
     function initialize(
         address _nftContract,
-        uint256 _tokenId,
         uint256 _maxPrice,
         uint256 _secondsToTimeout,
+        address[] calldata _deciders,
         Structs.AddressAndAmount calldata _split,
         Structs.AddressAndAmount calldata _tokenGate,
         string memory _name,
         string memory _symbol
     ) external initializer {
-        // validate maxPrice
-        require(_maxPrice > 0, "PartyBuy::initialize: must set price higher than 0");
         // initialize & validate shared Party variables
-        __Party_init(_nftContract, _tokenId, _split, _tokenGate, _name, _symbol);
+        __Party_init(_nftContract, _split, _tokenGate, _name, _symbol);
```

Comment:
can we emit an event with the deciders from the party contract itself so we don't have to query the factory to find out who the deciders are?

Reply:
it's my understanding that we always query the Factory contract to get the original event to get the information about a Party. the Factory is the source of truth because a maliciously implemented Party impersonator could emit false information. I was pretty sure the workflow for the frontend had a dependency on querying the events anyway, but please correct me if I'm wrong @steveklebanoff

Reply:
the party address is an indexed param in the Factory event, and there should only ever be one event emitted with that party address. so you should be passing a filter to the Factory event query for that party address. then it will return one event only, so you don't have to iterate.

is there a reason it is difficult to query for an indexed param?

Reply:
if so, I can emit the event, but the redundancy is a bit of a bummer from a purist perspective

Reply:
also, again, a malicious imitation party could emit a fake deciders event -- the registered Factory address which we know is a better trusted source of truth

Reply:
but i'm open to you saying "you're being a stickler just do the thing"

Reply:
i would really prefer an event emitted from this contract: consider us hitting `/buy/[partyAddress]` and we're in a future where we have 5 party buy factories.  we'd have to loop thru each of the 5 factories to find the event where the party was deployed to get the deciders.  this is a lot different than just querying the address given

Reply:
we want deciders to be a map to quickly look up though -- hesitant to add the overhead of iterating thru an array

Reply:
https://github.com/PartyDAO/partybuy/pull/4/files#diff-ec6aa7226e6ca449b8e0595869cac0d0caaa58121b968081dc2b1ac156adc820R16

Is this the event in question?  It looks like the address is not indexed so it would be difficult to filter on.  However if it was indexed then 

> we'd have to loop thru each of the 5 factories to find the event where the party was deployed

Would no longer be true because we could filter for the event without specifying a Factory address and it would just return the one event and tell you the factory and decider info